app.py

```diff
@@ -265,9 +281,43 @@ def cli(
     _LOGGER.debug("Debug mode turned on")
     _LOGGER.info(f"Package releases version: %r", __version__)
 
+    if package_names_file_jsonpath and not package_names_file:
+        raise ValueError("Cannot use JSON path when no package names file provided")
+
+    if not package_names_file_jsonpath and package_names_file:
+        raise ValueError("JSON path required when obtaining package names from a JSON file")
+
+    if only_if_package_seen and package_names_file:
+        raise ValueError(
+            "Only one of --package-names-file and --only-if-packages-seen is "
+            "allowed method for describing packages to be checked"
+        )
+
+    package_names = None
+    if package_names_file:
+        with open(package_names_file, "r") as f:
+            content = yaml.safe_load(f)
+
+        package_names = []
+        for item in parse(package_names_file_jsonpath).find(content):
+            for entry in item.value:
+                if not isinstance(entry, str):
+                    raise ValueError(f"Found item in the file is not a string describing package name: {item!r}")
+                package_names.append(entry)
+
+        if not package_names:
+            _LOGGER.warning("No packages to be checked for new releases")
+            sys.exit(2)
```

Comment:
@pacospace Would this fit into the argo workflow?

Reply:
It should fit!! Thanks @fridex !!